src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -598,6 +607,11 @@ private Status TryScanMethod(MethodIL methodIL, Value[] parameters, Stack<Method
                                 return Status.Fail(methodIL.OwningMethod, opcode, "Reference field");
                             }
 
+                            if (field.FieldType.IsByRef)
+                            {
+                                return Status.Fail(methodIL.OwningMethod, opcode, "Byref field");
```

Comment:
This is not strictly necessary because we would throw/catch exception later, but ideally we should only do throw/catch on invalid programs.